### DISCUSSION TITLE
[Feature] PDF File Support

### DISCUSSION THREAD

**Discussion by UnknownAXHF** [immich-app/immich#8653] [Feature Request]
### I have searched the existing feature requests to make sure this is not a duplicate request.

- [X] Yes

### The feature

Hi, I'm wondering if it'd ever be possible to support PDFs files.

I do a lot of research and have images which I use your application for but some of these images come in multipage PDFs. I have looked at just converting them, but wondered if PDF support has ever been looked at.

Thank you

### Platform

- [X] Server
- [ ] Web
- [ ] Mobile

**Comment by oldwang12** [comment]
The Livp format is not yet supported.

**Comment by sancelot** [comment]
what you are looking for is called a DAM : Digital Asset Management

**Reply by gregmuellegger** [reply]
I looked for this feature as well and benefit from having PDF support - for me the usecase are **Memorabilia**. Like Photos there are other things that might be worth storing in a timeline which better fit next to images than in a digital asset management.

The timeline is kind of a visual replay of experienced moments. And sometimes PDFs fit in this. A few examples:

- Invitations or thank-you cards for weddings or other friend and familiy events
- A leaflet of a museum exhibition visited, which should fit into the photos taken at the museum
- Tickets for a concert, also related to the photos taken at the venue
- Postcards received. One page the visual frontend, second page the written content on the back

**Reply by gregmuellegger** [reply]
Here are a few scripts I use to store memorabilia in Immich: https://gist.github.com/gregmuellegger/3699d8ffb26ea39fb617c6e153f1775f

My workflow is:

- Convert from PDF to JPG with `pdf2jpg-stitch.sh "2025-01-01 Invitationcard.pdf"`
- Then set metadata as appropriate:
  - Set date from filename (I usually stick to YYYY-MM-DD as prefix to the filename for convention): `exifdate-from-filename.sh "2025-01-01 Invitationcard.jpg"`
  - Extract the text with OCR to the Exif description: `ocr-to-exif.sh "2025-01-01 Invitationcard.jpg"`
  - Add geolocation using the free OSM API: `geocode-image.sh "2025-01-01 Invitationcard.jpg" -l "Timesquare, New York"`
- Upload to immich

**Comment by bo0tzz** [comment]
I'm not sure why this is still open, but PDF support is definitely out of scope for Immich so I'll be closing this request.

**Comment by kevinisaac** [comment]
I definitely think Immich should consider supporting pdf and other miscellaneous files - not as primary media formats like videos and photos but as second class citizens at least.

I've been using Immich for a while now and it's great. But it would make my experience complete if it just allows me to store some pdf docs that I have on my device along with my photos and images so that I can pull them out using the fantastic AI search feature Immich already has.

I would actually settle for Immich allowing external plugins or something to do the job, but we'd still need official support for it on the mobile apps.

**Comment by omidabduli** [comment]
I think supporting pdf would be an amazing option. it can be like only the first pages not the whole book of cours
